htlcswitch/interceptable_switch.go

```diff
@@ -337,6 +342,18 @@ func (s *InterceptableSwitch) interceptForward(packet *htlcPacket,
 			htlcSwitch: s.htlcSwitch,
 		}
 
+		// Handle forwards that are too close to expiry.
+		handled, err := s.handleExpired(intercepted)
+		if err != nil {
+			log.Errorf("Error handling intercepted htlc "+
+				"that expires too soon: circuit=%v, "+
+				"incoming_timeout=%v, err=%v",
+				packet.inKey(), packet.incomingTimeout, err)
```

Comment:
Do we want to proceed here if we errored out/did not handle (ie `FailWithCode` errored in `handleExpired`), shouldn't we return false?

Reply:
Good one, I indeed forgot a return statement there.

Interesting that you say return false, because I was thinking of returning true. With return false, the packet is passed on to the switch as normal and this violates the always-on requirement, if that is configured. Returning true isn't ideal either, because the htlc will remain stuck.

Maybe this is all hypothetical, because there is no way for an error to happen unless there is a bug in the code?

Reply:
In the hypothetical case, isn't it better to release to the switch and let it handle the expiry accordingly? Chances are that there's nowhere to forward it onwards (since generally there's no next hop when intercepting), so it'll fail back anyway?

> this violates the always-on requirement

always on vs stuck htlc, difficult call

Reply:
Whether there is nothing to forward onwards depends on the application I think. For external invoice handling it is true, but I think the interceptor is also used for the interception of regular routed traffic?

I think that one can come up with a high stakes htlc case where a stuck htlc and force-closed channel is preferable to skipping the interceptor.

But as this error path should be possible in theory only, I am fine with returning false. Updated.